src/sentry/web/frontend/restore_organization.py

```diff
@@ -28,20 +30,22 @@ class RestoreOrganizationView(OrganizationView):
     required_scope = "org:admin"
     sudo_required = True
 
-    def get_active_organization(self, request: Request, organization_slug):
-        # A simply version than what comes from the base
+    def determine_active_organization(self, request: Request, organization_slug=None) -> NoReturn:
+        # A simplified version than what comes from the base
         # OrganizationView. We need to grab an organization
         # that is in any state, not just VISIBLE.
-        organizations = Organization.objects.get_for_user(user=request.user, only_visible=False)
+        organizations = organization_service.get_organizations(
+            user_id=request.user.id, scope=None, only_visible=False
+        )
 
-        try:
-            return next(o for o in organizations if o.slug == organization_slug)
-        except StopIteration:
-            return None
+        for o in organizations:
+            if o.slug == organization_slug:
+                self.active_organization = o
+                break
```

Comment:
@corps should `self.active_organization = None` after the for loop if no organization is found?

Reply:
This was addressed, yup.